newIDE/app/src/EventsFunctionsList/index.js

```diff
@@ -123,6 +133,54 @@ export default class EventsFunctionsList extends React.Component<Props, State> {
     );
   };
 
+  _getFunctionThumbnail = (eventsFunction: ?gdEventsFunction) => {
```

Comment:
You can move this function out of the class to lighten re-renders.

Reply:
This is neither changing state nor calling props callbacks that may in turn trigger a rerender higher up, so this shouldn't trigger any rerender already.